contracts/0.8.9/WithdrawalRequestNFT.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>, OpenZeppelin
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.8.9;
+
+import {IERC721} from "@openzeppelin/contracts-v4.4/token/ERC721/IERC721.sol";
+import {IERC721Receiver} from "@openzeppelin/contracts-v4.4/token/ERC721/IERC721Receiver.sol";
+import {IERC165} from "@openzeppelin/contracts-v4.4/utils/introspection/IERC165.sol";
+
+import {Strings} from "@openzeppelin/contracts-v4.4/utils/Strings.sol";
+import {EnumerableSet} from "@openzeppelin/contracts-v4.4/utils/structs/EnumerableSet.sol";
+import {Address} from "@openzeppelin/contracts-v4.4/utils/Address.sol";
+
+import {IWstETH, WithdrawalQueue} from "./WithdrawalQueue.sol";
+import {AccessControlEnumerable} from "./utils/access/AccessControlEnumerable.sol";
+import {UnstructuredRefStorage} from "./lib/UnstructuredRefStorage.sol";
+
+/// @title NFT implementation on top of {WithdrawalQueue}
+/// NFT is minted on every request and burned on claim
+///
+/// @author psirex, folkyatina
+contract WithdrawalRequestNFT is IERC721, WithdrawalQueue {
```

Comment:
let's add metadata (*open the gate*)
and special lever to update it on the go :brain: